### ISSUE TITLE
add Softmax layer to stax

### ISSUE THREAD

**Issue by cookfish** [google/jax#182]
test in examples/mnist_classifier.py,
```python
def accuracy(params, batch):    
  inputs, targets = batch    
  target_class = np.argmax(targets, axis=1)
  x = predict(params, inputs)
  print(x[0])
```
the final layer is LogSoftmax, but the output seems not correct, 
```python  
Starting training...
[-2.3113673 -2.6440005 -2.4797316 -1.79847   -1.6207608 -2.931935
 -3.303906  -2.7275395 -2.2099946 -2.1403143]
```

**Comment by mattjj** [comment]
Can you say a bit more about what might be incorrect here? The sum of the elementwise-exp of those numbers is 1, which is intended.

**Comment by cookfish** [comment]
I mean the sum of those numbers should be 1, not their elementwise-exp's.

**Comment by cookfish** [comment]
sorry, i misunderstood log-softmax, maybe it will be more convenient if providing loss functions.

**Comment by mattjj** [comment]
Got it! I'll add that to my todo list.
